dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/CollectionRestRepository.java

```diff
@@ -174,19 +175,26 @@ public CollectionRest findOne(Context context, UUID id) {
                     CommunityRest.CATEGORY + "." + CommunityRest.NAME + " with id: " + communityUuid
                         + " not found");
             }
-            List<Collection> collections = cs.findAuthorized(context, com, Constants.ADD);
-            return converter.toRestPage(collections, pageable, utils.obtainProjection());
-        } catch (SQLException e) {
+            List<Collection> collections = cs.findAuthorizedCollectionsInSOLR(q, context, com,
+                                              Math.toIntExact(pageable.getOffset()),
+                                              Math.toIntExact(pageable.getOffset() + pageable.getPageSize()));
+            int tot = cs.countAuthorizedCollectionsInSOLR(q, context, com);
+            return converter.toRestPage(collections, pageable, tot , utils.obtainProjection());
+        } catch (SQLException | SearchServiceException e) {
             throw new RuntimeException(e.getMessage(), e);
         }
     }
 
     @SearchRestMethod(name = "findAuthorized")
```

Comment:
I'm just now realizing this `@SearchRestMethod` seems mis-named as well.  We are *not* finding all Collections you are authorized to Read or edit or similar.  This is very specific to collections you can *submit to*.  Shouldn't this be called `findSubmitAuthorized`  or `findAuthorizedForSubmit`?